core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -363,24 +387,31 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
             }
         });
 
-        return mapAcknowledgementFutures(futuresMap);
+        return mapAcknowledgementFutures(futuresMap, Optional.empty());
     }
 
-    private CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> mapAcknowledgementFutures(Map<TopicIdPartition, CompletableFuture<Throwable>> futuresMap) {
+    private CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.PartitionData>> mapAcknowledgementFutures(
+        Map<TopicIdPartition, CompletableFuture<Throwable>> futuresMap,
+        Optional<Consumer<Collection<String>>> metricsHandler
```

Comment:
This really is the `failedMetricsHandler`

Reply:
Done.